### ISSUE TITLE
Unique ID - Not Unique

### ISSUE THREAD

**Issue by merlin051** [bwssytems/ha-bridge#513]
When using Edit/Copy device the unique ID is not unique.

I'm on latest version 4.2.0.

**Comment by bwssytems** [comment]
Minor, but may impact applications that expect the HUE Unique ID to be Unique....

**Comment by bwssytems** [comment]
Also, I updated mapId to add a "-copy" to the end of it.

**Comment by Carl-Horst** [comment]
Seems to be the case with Alexa. After Editing the device cannot be Switches Any more by alexa.

**Comment by merlin051** [comment]
I had that problem, also assumed it's the duplicate ID.

I had to restart the service completely, the re-initialize didn't work for me.

**Comment by mol7os** [comment]
After many time of research it looks like many issues are depending to the Unique ID handling in the HABridge.

The UniqueID is not really unique. Renumbering devices will start to pin the UniqueID to new devices. Alexa itself finds the devices but sometimes different named with different HABridge IDs.

Old devices are listed and mess up "Alexa". The only common denominator of the ghost devices is the UniqueID. 

Example: 
Device A HAID: 1 UID: 00:17:88:5E:D3:64-00
Device B HAID: 2 UID: 00:17:88:5E:D3:65-00
Device C HAID: 3 UID: 00:17:88:5E:D3:66-00
Alexa device count: 3

Renumbered: 
Device B HAID: 1 UID: 00:17:88:5E:D3:64-00
Device C HAID: 2 UID: 00:17:88:5E:D3:65-00
Device A HAID: 3 UID: 00:17:88:5E:D3:66-00
Alexa device count: 6 (old devices will be found, same as new ordered)

Device deleted and renumbered:
Device B HAID: 1 UID: 00:17:88:5E:D3:64-00
Device X HAID: 2 UID: 00:17:88:5E:D3:65-00
Device A HAID: 3 UID: 00:17:88:5E:D3:66-00
Alexa device count: 7 (old devices will be found, same as new ordered)

Maybe it makes sense to take a look at the UniqueID handling again.

**Comment by bwssytems** [comment]
you can renumber devices and set the starting device number in bridge control to make this more unique.

**Comment by mol7os** [comment]
Sure this will mix the UID´s but the Alexa devices will find them. "Ghost" devices or double devices cause by reachable UIDs.

**Comment by bwssytems** [comment]
So I think this still is an issue with devices not being purged properly from the echo. There have been different suggestions on how to do this. The best seems to delete individually on the echo web page not the app.

**Comment by mol7os** [comment]
None of the "device deletion" fixes will work. (Web App All, by hand or IOS App by hand)
Over many months I receive ghost devices. The first time with the newer Versions (5+) of HABridge and the new Echo2 devices.
I removed the HAbridge devices to a minimum cause of this (every scan mess up my whole system).

Renumbering, Deleting, Tracing, .... nothing worked for me. But I found out that THE UID which is normally a HUE unique ID for a lamp is never changed. Regardless of deletion or renumbering, all UID start again from UID: 00:17:88:5E:D3:64-00.

If this UID is cached by Alexa the discovery will find them all the time, ok different devices with the same UID.
